### ISSUE TITLE
Add support for DuckDNS

### ISSUE THREAD

**Issue by coreybruce** [http-party/http-server#777]
Hi I would love if this had a way to connect to my duckdns domain so I can have a instant working domain and SSL HTTPS, DuckDNS has a your domain you created and a required token you generate to work so if that could be added than that would add a great easy way to get a web server up. :)

**Comment by thornjad** [comment]
What is needed to support DuckDNS? Could you better detail how you'd use http-server for this and what's missing?

**Comment by coreybruce** [comment]
Sure so if you go to duckdns.org you will see you can login via GitHub, Google etc. Once you've done that (I logged in via Github) you can create a domain, if you could add support for DuckDNS like Nextcloudpi has and point to your domain and have a secure way to give it your security token that would make it so easy to add a secure Https domain to a http-sever instant.

**Comment by coreybruce** [comment]
Would this be possible?

**Comment by andrewcrook** [comment]
This could be done by a script with the Duck DNS REST API as a scheduled task I am sure there are plenty out there.

 I am not sure it needs a feature in http-server which is a simple HTTPS server and not intended to be used on the public Internet as such. You would also need to set up port forwarding on your router if on a private machine and network. 

However, would something like https://ngrok.com/ be better?  
its easy, fast, can fire up as and when needed and doesn’t require port forwarding etc.

**Comment by thornjad** [comment]
I agree this _may_ not be something in the scope of http-server but I'd welcome a PR showing how it can be done!

**Comment by coreybruce** [comment]
Not 100% sure on the technicals but I'd say do it the same way nextcloud does by entering the domain name and your token which would register to the domain.
